src/rendering/MaterialParser/MaterialParser.cc

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2023 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <cstddef>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "MaterialParser.hh"
+
+using namespace gz;
+using namespace sim;
+
+/////////////////////////////////////////////////
+MaterialParser::MaterialParser()
+  : configLoader()
+{
+}
+
+/////////////////////////////////////////////////
+void MaterialParser::Load()
+{
+  ConfigLoader::loadMaterialFiles(&this->configLoader);
+}
+
+/////////////////////////////////////////////////
+MaterialParser::MaterialValues MaterialParser::GetMaterialValues(
+  const std::string& material)
+{
+  MaterialValues values;
+  std::map<std::string, ConfigNode *> scripts =
+    this->configLoader.getAllConfigScripts();
+
+  std::map<std::string, ConfigNode *> ::iterator it;
+
+  for (it = scripts.begin(); it != scripts.end(); ++it) {
+    std::string name = it->first;
+    if (name.find(material) != std::string::npos) {
```

Comment:
Not sure what's going on here. Couldn't we just search for the material name with `scripts.find(material)` instead of having a for loop?  Also, should we return a `std::optional<MaterialValues>` so that if the material name is not found, we can emit a warning? This relates to my previous comment about the absence of ambient being used to determine if a material is found.

Reply:
`scripts` is a map with key of material names like "material Gazebo/Grey" and value of `ConfigNode`. I need to compare a substring "Gazebo/Grey" with the `scripts` keys, so there's a for loop needed. 

Updated to make `MaterialValues` optional.

Reply:
Okay, I guess in future versions, we'd just store "Gazebo/Grey" in the map, but it's okay for now.

Reply:
I think the value_or would provide a default in this case.